bpf/init.sh

```diff
@@ -180,16 +184,28 @@ function setup_proxy_rules()
 			if [ -z "$(ip -4 rule list $to_proxy_rulespec)" ]; then
 				ip -4 rule add $to_proxy_rulespec
 			fi
-			if [ -z "$(ip -4 rule list $from_ingress_rulespec)" ]; then
-				ip -4 rule add $from_ingress_rulespec
-			fi
+			case "${MODE}" in
+			"routed")
+				;;
+			*)
+				if [ -z "$(ip -4 rule list $from_ingress_rulespec)" ]; then
+					ip -4 rule add $from_ingress_rulespec
+				fi
+				;;
+			esac
 		fi
 
 		# Traffic to the host proxy is local
 		ip route replace table $TO_PROXY_RT_TABLE local 0.0.0.0/0 dev lo
 		# Traffic from ingress proxy goes to Cilium address space via the cilium host device
-		ip route replace table $PROXY_RT_TABLE $IP4_HOST/32 dev $HOST_DEV1
-		ip route replace table $PROXY_RT_TABLE default via $IP4_HOST
+		case "${MODE}" in
+		"routed")
+			;;
```

Comment:
This won't get rid of any existing rule of the form that we are skipping configuration of here. Maybe that's okay because users will only begin using this mode with the new release, but if someone changes from regular mode to routed mode then it won't clean up the rule that was configured in the normal mode. In particular this could affect the CI.

Reply:
(Resolved in the latest commits)